golem_messages/message/tasks.py

```diff
@@ -162,10 +166,62 @@ class SubtaskResultsAccepted(base.Message):
     ] + base.Message.__slots__
 
 
-class SubtaskResultsRejected(base.Message):
+class SubtaskResultsRejected(base.AbstractReasonMessage):
+    """
+    Message sent from the Requestor to the Provider, rejecting the provider's
+    completed task results
+
+    also, sent from the Concent to the Provider in case of additional
+    verification (when the verdict of the verification is negative)
+    or in case of forced results verdict (when the verdict is
+    negative - either because the work itself was deemed invalid by the
+    requestor
+    """
     TYPE = TASK_MSG_BASE + 11
 
-    __slots__ = ['subtask_id'] + base.Message.__slots__
+    __slots__ = [
+        'report_computed_task',
+        'force_get_task_result_failed',
+    ] + base.AbstractReasonMessage.__slots__
+
+    @enum.unique
+    class REASON(enum.Enum):
+        VerificationNegative = 'Results verification negative'
+        ConcentDuplicateRequest = 'Concent had already processed this request'
+        ConcentResourcesFailure = \
+            'Concent could not retrieve resources to verify'
+        ConcentVerificationNegative = 'Concent results verification negative'
+        ConcentMessageRejected = \
+            'Included computed task report was found corrupt'
+        ForcedResourcesFailure = \
+            'Concent reported failure to retrieve the resources to verify'
+        ResourcesFailure = \
+            'Could not retrieve resources'
+
+    def deserialize_slot(self, key, value):
+        # pylint: disable=cyclic-import
+        #
+        # with the current specification, it will be difficult to do without
+        # the cyclic import here since per-specs, the `SubtaskResultsRejected`
+        # message can contain the concent's `ForceGetTaskResultFailed` and
+        # at the same time, the concent messages also can contain
+        # `SubtaskResultsRejected` ...
+        #
+        from .concents import ForceGetTaskResultFailed
+        # pylint: enable=cyclic-import
+
+        value = super().deserialize_slot(key, value)
+        value = base.deserialize_verify(
+            key, value,
+            verify_key='report_computed_task',
+            verify_class=ReportComputedTask,
+        )
+        value = base.deserialize_verify(
+            key, value,
+            verify_key='force_get_task_result_failed',
+            verify_class=ForceGetTaskResultFailed,
+        )
```

Comment:
I would put this deserializer in `message.concents` as `deserialize_force_get_task_result_failed`

Reply:
@jiivan hmm... generally, I find those helpers of dubious value... you're essentially just binding the parameter name with the class name but otherwise, not adding much to the code readability and/or DRY... but sure, I'll add that helper...